apps/src/turtle/requiredBlocks.js

```diff
@@ -197,9 +197,7 @@ var defineWithArg = function (func_name, arg_name) {
       return (
         block.type === 'procedures_defnoreturn' &&
         block.getFieldValue('NAME') === func_name &&
-        block.parameterNames_ &&
-        block.parameterNames_.length &&
-        block.parameterNames_[0] === arg_name
+        block.getProcedureModel()?.getParameters()[0]?.getName() === arg_name
```

Comment:
could `getParameters` return an empty array or null and cause us an exception trying to access index 0?

Reply:
Good catch. Yes, it could. Added one more link to the chaining. :)